### ISSUE TITLE
Handle `fail_stop` / non teardown tasks in TASK SDK

### ISSUE THREAD

**Issue by amoghrajesh** [apache/airflow#44951]
### Body

Handle the part where we stop all remaining tasks for while marking a task as failed.
Usage: https://github.com/apache/airflow/blob/aaf29ee54bd8666a8dc3a129aed213f6b3b31bde/airflow/models/taskinstance.py#L3086-L3087

My thinking here is that we introduce the "fail_stop" property in the serialised dag and while executing a task (marking it running -- done in the /run endpoint in execution API), if that property is set, we just fail.

**Comment by kaxil** [comment]
Ths is still pending, right @amoghrajesh ? Or is this one already complete?

**Comment by amoghrajesh** [comment]
Yeah this is still pending and we are waiting on https://github.com/apache/airflow/pull/45327 to get this one in.

**Comment by Lee-W** [comment]
just merged https://github.com/apache/airflow/pull/45327

**Comment by ashb** [comment]
See also https://github.com/apache/airflow/issues/26854#issuecomment-2435388752

**Comment by prabhusneha** [comment]
I can work on this. Please assign it to me.

**Comment by amoghrajesh** [comment]
Sure, assigning.

**Comment by kaxil** [comment]
@prabhusneha @amoghrajesh Where are we on this?

**Comment by prabhusneha** [comment]
> @prabhusneha @amoghrajesh Where are we on this?

@kaxil  I've taken an initial look at it and planning on working on it in the coming week
